drivers/storage/portworx/component/security.go

```diff
@@ -89,11 +89,20 @@ func (c *security) Initialize(
 
 // IsEnabled checks if the components needs to be enabled based on the StorageCluster
 func (c *security) IsEnabled(cluster *corev1.StorageCluster) bool {
-	return pxutil.SecurityEnabled(cluster)
+	return pxutil.AuthEnabled(&cluster.Spec) || pxutil.IsTLSEnabledOnCluster(&cluster.Spec)
```

Comment:
> I think we should make TLS a separate component to keep it simple and call this component as Auth instead of Security. It would be only needed when we start auto-generating certs.

As discussed in slack: renamed `security` --> `auth`. Did not create a `tls' component because it would be empty right now. Will add it in the future when we do auto-generating tls certs and there is work for tls component.